decode.go

```diff
@@ -23,8 +23,11 @@ type Decoder struct {
 // The decoder introduces its own buffering and may read data from r beyond
 // the logfmt records requested.
 func NewDecoder(r io.Reader) *Decoder {
+	buf := make([]byte, 0, 64*1024)
+	scanner := bufio.NewScanner(r)
+	scanner.Buffer(buf, 1024*1024)
```

Comment:
I would rather not make this the default behavior. I also think we should make it more flexible by allowing the caller to set the max line length instead of picking an arbitrary size that is too large for most uses and might still be too small in some cases. I suggest adding a new function `NewDecoderSize(r io.Reader, size int) *Decoder` instead of modifying the existing `NewDecoder` function.